### PR TITLE
Add API to fetch tileable graph as JSON

### LINKED ISSUE #2235: API endpoint for tileable graph
an api that collects and sends information of tileable graphs can help us display the status/make data visualization of tileable graphs on mars web

### PR DESCRIPTION
**Pull request by RandomY-2**
## What do these changes do?

API endpoints that collect information on the tileable graphs created from each execution. The endpoint sends the tileables and the dependencies(edges) in the graph.

## Related issue number

Fixes #2235

### CODE REVIEW

**Comment by wjsi** [comment]
Why you need APIs for every tileable? When we need to display the progress of the task, you only need one graph containing all tileables in the task. Note that graphs extracted from different tileables may overlap. For instance, in the task

```python
inp = mt.ones(10, 10)
left = inp.sum()
right = inp.product()
mars.execute(left, right)
```

graph for tileable `left` overlaps with graph for tileable `right` on node `TensorOnes`.

File: mars/services/task/supervisor/processor.py [hunk 1]
```diff
@@ -463,6 +493,12 @@ def get_result_tileables(self):
             result.append(build_fetch(tiled))
         return result
 
+    def get_result_tileable_graph(self):
```
Comment (wjsi) [inline mars/services/task/supervisor/processor.py:496]: Rename as `get_tileable_graph_as_dict` as the original name may imply that the return value is a `TileableGraph`.

File: mars/services/task/supervisor/processor.py [hunk 2]
```diff
@@ -50,6 +50,36 @@ async def inner(processor: "TaskProcessor", *args, **kwargs):
 
     return inner
 
+def format_graph_data(graph: TileableGraph):
```
Comment (wjsi) [inline mars/services/task/supervisor/processor.py:53]: Maybe it is better to put this code block inside the method implementation, as it is not be used elsewhere.

File: mars/services/task/api/web.py [hunk 3]
```diff
@@ -103,6 +104,12 @@ async def get_task_result(self, session_id: str, task_id: str):
         res = await oscar_api.get_task_result(task_id)
         self.write(json.dumps(_json_serial_task_result(res)))
 
+    @web_api('(?P<task_id>[^/]+)' + '/tileable_graph', method='get', arg_filter={'action': 'get_tileable_graph'})
```
Comment (wjsi) [inline mars/services/task/api/web.py:107]: Rename the action as `get_tileable_graph_as_json`?

File: mars/services/task/api/web.py [hunk 4]
```diff
@@ -103,6 +104,12 @@ async def get_task_result(self, session_id: str, task_id: str):
...
+    async def get_tileable_detail_by_key(self, session_id: str, task_id: str):
```
Comment (wjsi) [inline mars/services/task/api/web.py:108]: Make the name of the method consistent with the name of the action.

File: mars/services/task/api/oscar.py [hunk 5]
```diff
@@ -111,6 +111,11 @@ async def submit_tileable_graph(self,
         except mo.ActorNotExist:
             raise RuntimeError('Session closed already')
 
+    async def get_tileable_detail_by_key(self, task_id: str):
```
Comment (wjsi) [inline mars/services/task/api/oscar.py:114]: Ditto.

File: mars/services/task/supervisor/manager.py [hunk 6]
```diff
@@ -171,6 +173,14 @@ async def submit_tileable_graph(self,
 
         return task_id
 
+    async def get_tileable_detail_by_key(self, task_id):
```
Comment (wjsi) [inline mars/services/task/supervisor/manager.py:176]: Ditto.

File: mars/services/task/api/oscar.py [hunk 7]
```diff
@@ -111,6 +111,11 @@ async def submit_tileable_graph(self,
...
+        return await self._task_manager_ref.get_tileable_detail_by_key(
```
Comment (wjsi) [inline mars/services/task/api/oscar.py:115]: Ditto.

**Review by wjsi — COMMENTED** [review]
LGTM in general, I proposed some renames. Also note that we require every code file with a blank line at end. You may change your project settings. More code style issues can be seen at https://github.com/mars-project/mars/pull/2253/checks?check_run_id=3182620201. All these errors and warnings need to be fixed before merge.

File: mars/services/web/ui/src/task_info/TaskListPage.js [hunk 8]
```diff
@@ -32,18 +32,36 @@ class TaskList extends React.Component {
         this.state = {};
     }
 
+    fetchTilebaleGraph() {
```
Comment (wjsi) [inline mars/services/web/ui/src/task_info/TaskListPage.js:35]: Is this function added only for test purpose? If so, add it later when we actually implements task visualization frontend.

**Comment by qinxuye** [comment]
Flake8 failed, refer to https://github.com/mars-project/mars/pull/2253/checks?check_run_id=3193310180

**Review by qinxuye — APPROVED** [review]
LGTM

**Review by wjsi — APPROVED** [review]
LGTM

**Comment by qinxuye** [comment]
Thanks for your contribution, look forward to seeing more contributions from you.

**Pull request merged**